### PR TITLE
add fix for e2e script not running currectly on Mojave: feature/18170

### PR DESCRIPTION
**Pull request by accbjt**
## Description
During the process of trying to write documents on unit tests, some issues were found on the e2e test while running locally.

Nightwatch script wasn't running command async, so the mock-api server on port 3000 would wait till it stopped to go further to with running the tests. It should run asynchronously. The way to solve it was to add a & after the mock-api script

The mock-api shell script sleeps after 5 seconds which will break the tests because the mock-api is no longer running, which makes the tests fail after 5 seconds.

Device: Mac
OS: Mojave
Browser: Chrome

Steps to test
- Run `$ yarn test:e2e`
- While the test is running run `$ lsof -i tcp:3000` to check if the mock-api is running
- after the test is done run `$ lsof -i tcp:3000` and make sure port 3000 stopped running

## Testing done
Local Testing

## Acceptance criteria
- [ ] e2e test runs properly and doesn't get stuck
- [ ] mock-api server should stay running on port 3000 during the whole test and should shut off when finished.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: script/run-mockapi.sh [hunk 1]
```diff
@@ -7,7 +7,8 @@ trap 'if [ $(jobs -p) ] ; then kill $(jobs -p); fi' EXIT
 if [ "$(nc -z localhost 3000; echo $?)" -ne 0 ]; then
     echo "Starting mockapi.js..."
     node src/platform/testing/e2e/mockapi.js &
-    sleep 5;
+    # waits for process to stop
+    wait;
```
Comment (accbjt) [inline script/run-mockapi.sh]: `sleep` was causing the mock-api to stop before the tests finish. `wait` should just wait for the process to end to stop the server.

File: script/run-nightwatch.sh [hunk 2]
```diff
@@ -29,7 +29,7 @@ trap 'if [ $(jobs -p) ] ; then kill $(jobs -p); fi' EXIT
 
 BUILDTYPE=${BUILDTYPE:-vagovdev}
 
-"$(dirname "$0")"/run-mockapi.sh
+"$(dirname "$0")"/run-mockapi.sh &
```
Comment (accbjt) [inline script/run-nightwatch.sh]: adding the `&` because this didn't run asynchronously and would pause. This will make it asynchronously.

**Comment by jbalboni** [comment]
@accbjt This works for me (a non-Mojave Mac user), I'd like to get someone on Linux (e.g. @cvalarida) to try it out, too.

**Comment by ncksllvn** [comment]
@accbjt Could this be the same/related issue as described here, https://github.com/department-of-veterans-affairs/vets.gov-team/issues/17912?

**Comment by accbjt** [comment]
@ncksllvn Yeah this looks like the same issue. This PR should fix the problem.

**Review by ncksllvn — APPROVED** [review]
LGTM! `npm run test:e2e` is restored 🎉 

CC: @cvalarida

**Comment by cvalarida** [comment]
For posterity, it doesn't kill the mock API when the tests are over on my linux machine. This has been a problem, though, so it's not due to this PR. Other than that, it works great. :+1:

**Pull request merged**